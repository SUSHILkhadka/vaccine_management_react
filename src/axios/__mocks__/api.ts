import { vaccineArray } from '../../__test__/utils/sort.test';

module.exports = {
  post: (url: string, body: any = "without any body") => {
    return {
      data: {
        data: body,
      },
    };
  },
  get: (url: string) => {
    return {
      data: {
        data: vaccineArray,
      },
    };
  },
  put: (url: string, body: any) => {
    return {
      data: {
        data: body,
      },
    };
  },
  delete: (url: string, body: any = "without any body") => {
    return {
      data: {
        data: body,
      },
    };
  },
};
