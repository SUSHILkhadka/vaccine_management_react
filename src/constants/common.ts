export const URL_TO_BACKEND='http://localhost:8000'
// export const imageURL="https://images.unsplash.com/profile-1446404465118-3a53b909cc82?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32&s=a2f8c40e39b8dfee1534eb32acfa6bc7"
// export const imageURL="https://xsgames.co/randomusers/avatar.php?g=female"
export const imageURL="https://www.gravatar.com/avatar/2c7d99fe281ecd3bcd65ab915bac6dd5?s=250"
export const IMAGE_COMPRESSION_PERCENTAGE=50;