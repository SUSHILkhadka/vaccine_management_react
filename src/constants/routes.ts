export const PATH_ROOT = '/';
export const PATH_LOGIN = '/login';
export const PATH_REGISTER = '/register';
export const PATH_SETTING = '/setting';
export const PATH_VACCINE_ADD = '/vaccine/add';
export const PATH_VACCINE_TABLE = '/';
export const PATH_VACCINE_EDIT = '/vaccine/edit';
