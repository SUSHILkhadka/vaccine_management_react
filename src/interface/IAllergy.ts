export type IAllergy = {
    id: number;
    name: string;
    status: 'read' | 'added' | 'edited' | 'deleted';
  };
  