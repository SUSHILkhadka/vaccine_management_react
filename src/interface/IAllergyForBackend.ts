export interface IAllergyForBackend {
  id: number;
  name: string;
  vaccineId: number;
}
export type IAllergyForBackendToInsert = Omit<IAllergyForBackend, 'id'>;
