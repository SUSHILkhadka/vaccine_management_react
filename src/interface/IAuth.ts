export interface IAuth {
    id: number;
    username: string;
    email: string;
    status?: 'loading' | 'fulfilled' | 'rejected';
  }
  