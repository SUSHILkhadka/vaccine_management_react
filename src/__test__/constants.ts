import { IAllergy } from '../interface/IAllergy';
import { IVaccine } from '../interface/IVaccine';

export const vaccine: IVaccine = {
  id: 24,
  name: 'FluMist',
  description: 'Used in Seasonal Flu. Its other name is LAIV.',
  numberOfDoses: 2,
  releaseDate: '2008-01-30',
  photoUrl: '',
  isMandatory: false,
};

export const allergyList: IAllergy[] = [
  {
    id: -1,
    name: 'allery1',
    status: 'added',
  },
  {
    id: 3,
    name: 'allery3',
    status: 'added',
  },
  {
    id: 3,
    name: 'allery3',
    status: 'deleted',
  },
];

export const vaccineArray = [
  {
    id: 1,
    name: 'a',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: false,
  },
  {
    id: 2,
    name: 'vaccinenameiswired',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: true,
  },
  {
    id: 3,
    name: 'a',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: false,
  },
  {
    id: 4,
    name: 'a',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: true,
  },
];
