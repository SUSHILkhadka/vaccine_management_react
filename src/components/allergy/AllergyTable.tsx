import { useSelector } from 'react-redux';
import { IAllergy } from '../../interface/IAllergy';
import { RootState } from '../../redux_toolkit/stores/store';
import './Allergy.scss';
import AllergyCard from './AllergyCard';
import AllergyAddButton from './AlleryAddButton';

const AllergyTable = () => {
  const allergyArrayInfo = useSelector((state: RootState) => state.allergy);
  return (
    <div>
      <AllergyAddButton />
      <div className='allergy--list--container'>
        <>
          {allergyArrayInfo.map((element: IAllergy, index: number) => {
            if (element.status != 'deleted')
              return <AllergyCard key={element.id} index={index} name={element.name} />;
          })}
        </>
      </div>
    </div>
  );
};
export default AllergyTable;
