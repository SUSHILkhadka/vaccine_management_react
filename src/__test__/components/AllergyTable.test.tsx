import AllergyTable from '../../components/allergy/AllergyTable';
import { IAllergy } from '../../interface/IAllergy';
import '../../__mocks__/matchMedia';
import { allergyList } from '../constants';
import { renderWithProvidersAndNavigation } from './customRender';


it('should render allergy table page properly', () => {
  const page = renderWithProvidersAndNavigation(<AllergyTable />, {
    preloadedState: {
      allergy: allergyList,
    },
  });

  expect(page.asFragment()).toMatchSnapshot();
});
