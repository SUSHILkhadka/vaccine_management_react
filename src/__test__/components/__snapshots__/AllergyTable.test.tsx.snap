// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render allergy table page properly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <button
        class="ant-btn ant-btn-default button button--save"
        type="button"
      >
        <span>
          Add allergy
        </span>
      </button>
      <div />
    </div>
    <div
      class="allergy--list--container"
    >
      <div>
        <div
          class="allergy--card"
        >
          allery1
          <div
            class="allergy--card--deleteicon"
          >
            ✕
          </div>
        </div>
        <div />
      </div>
      <div>
        <div
          class="allergy--card"
        >
          allery3
          <div
            class="allergy--card--deleteicon"
          >
            ✕
          </div>
        </div>
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
